src/online2bin/online2-tcp-nnet3-decode-faster.cc

```diff
@@ -431,7 +431,7 @@ bool TcpServer::ReadChunk(size_t len) {
       KALDI_WARN << "Socket timeout! Disconnecting...";
       break;
     }
-    if (client_set_[0].revents != POLLIN) {
```

Comment:
@hhadian what do you think about this?

Reply:
It seems good to me. At least AFAIK `revents` is a bit mask and should not be compared like this (e.g. other bits in addition to POLLIN might be set). Maybe it's better to keep this if-block too but like this: `if (!(client_set_[0].revents & POLLIN))`.
Babak, can you please see if this will work too in your tests?
Anyway more checks might be needed, I don't remember all details right now (see http://man7.org/linux/man-pages/man2/poll.2.html)

Reply:
@hhadian Yes it works too.